### ISSUE TITLE
Procedure Not Used Inspection - False Positive

### ISSUE THREAD

**Issue by SmileyFtW** [rubberduck-vba/Rubberduck#5817]
**Rubberduck version information**
Version 2.5.2.5951
OS: Microsoft Windows NT 10.0.19042.0, x64
Host Product: Microsoft Office x86
Host Version: 16.0.14026.20246
Host Executable: EXCEL.EXE

**Description**
From a Factory Method a `Let` is flagged as not being used when in fact it is. All other `Let`ters/`Get`ters in the Factory are not flagged, yet used in the same way.

**To Reproduce**
1. I have no idea how to reproduce it.
2. 
**Expected behavior**
The `Let`ter would not be flagged as not used

**Screenshots**
![image](https://user-images.githubusercontent.com/42221335/122773411-f5adac00-d26d-11eb-8703-be6bee9ee3fd.png)
![image](https://user-images.githubusercontent.com/42221335/122773655-255cb400-d26e-11eb-8651-4f6170c5fd5a.png)

**Logfile**
[RubberduckLog.txt](https://github.com/rubberduck-vba/Rubberduck/files/6687082/RubberduckLog.txt)

**Additional context**

**Comment by retailcoder** [comment]
Could it be just the `With` block throwing a wrench here?

**Comment by SmileyFtW** [comment]
I don’t think so because there are other items in the With block, regardless of Type, that aren’t throwing the exception.

**Comment by BZngr** [comment]
In the process of trying to resolve this one - found that the fundamental issue is that the `Parser` returns 0 rather than 1 `IdentifierReference(s)` for `Property Let Description` in the MCVE below.  So, the inspection logic is not the issue, it is correctly evaluating the _incorrect_ data it has received.  So, it does seem that the `With` statement is "throwing a wrench" into this scenario.  

Since I steer clear of anything involving the `Parser`, I'll leave behind the MCVE test code I was using just in case it might save a little bit of time for the next person investigating this one.

```
[Test]
[Category("Inspections")]
public void MCVE()
{
     var inputCode =
$@"
Option Explicit

Private Type T{MockVbeBuilder.TestModuleName}
    Description As String
End Type

Private this As T{MockVbeBuilder.TestModuleName}

Public Function Create(ByVal pArg As String) As {MockVbeBuilder.TestModuleName}
    Dim tm As {MockVbeBuilder.TestModuleName}
    Set tm = New {MockVbeBuilder.TestModuleName}
    With tm 
        .Description = pArg
    End With
    Set Create = tm 
End Function

Public Property Let Description(ByVal RHS As String)
    this.Description = RHS
End Property
";

     var modules = new (string, string, ComponentType)[]
     {
            (MockVbeBuilder.TestModuleName, inputCode, ComponentType.ClassModule),
     };

     var results = InspectionResultsForModules(modules);

     Assert.AreEqual(0, InspectionResultsForModules(modules).Count(result=>
         result.Target.DeclarationType.HasFlag(DeclarationType.PropertyLet)));
}
```

**Comment by MDoerner** [comment]
This might actually have the same root cause as #5991. I will have a look. Thanks for the MCVE.

**Comment by BZngr** [comment]
Welp...I too am at a loss.  I've just re-run the MCVE both as a VS test and as code in Excel...with no inspection result!  It was _very_ repeatable when I commented 2 days ago.
